src/tidecli/models/task_data.py

```diff
@@ -111,7 +128,23 @@ def pretty_print(self) -> str:
 
     def to_json(self) -> dict:
         """Convert to dict."""
-        task_data = self.dict()
-        task_data["task_files"] = [task_file.dict() for task_file in self.task_files]
+        task_data = self.model_dump()
+        task_data["task_files"] = [task_file.model_dump() for task_file in self.task_files]
 
         return task_data
+
+
+class TideWeekData(BaseModel):
```

Comment:
TideCoursePartData would be a better name imo. Not all courses are split into weeks.

Reply:
Muutettu